guess_the_number_project/main.py

```diff
@@ -0,0 +1,46 @@
+from random import randint
+from art import logo
+
+EASY_LEVEL_TURNS = 10
+HARD_LEVEL_TURNS = 5
+
+def check_answer(guess, answer, turns):
+  """checks answer against guess. Returns the number of turns remaining."""
+  if guess > answer:
+    print("Too high.")
+    return turns - 1
+  elif guess < answer:
+    print("Too low.")
+    return turns - 1
+  else:
+    print(f"You got it! The answer was {answer}.")
+
+def set_difficulty():
+  level = input("Choose a difficulty. Type 'easy' or 'hard': ")
+  if level == "easy":
+    return EASY_LEVEL_TURNS
+  else:
+    return HARD_LEVEL_TURNS
+
+def game():
+  print(logo)
+  print("Welcome to the Number Guessing Game!")
+  print("I'm thinking of a number between 1 and 100.")
+  answer = randint(1, 100)
+
+  turns = set_difficulty()
+  guess = 0
+  while guess != answer:
+    print(f"You have {turns} attempts remaining to guess the number.")
+
+    guess = int(input("Make a guess: "))
+    turns = check_answer(guess, answer, turns)
+    if turns == 0:
+      print("You've run out of guesses, you lose.")
+      return
```

Comment:
Do you understand what's happening right here? Put another way -- why does the game end if you win?

Reply:
because in the game mechanics if you used all the turns you will lose. should I put it in the while loop so after win or lose it will be back to the start of the game?

Reply:
Nope, the design is good! I see that you understand why it ends if you run out of turns, but how does the program "know" to end if you get it right (and you're not out of turns yet)?